service/ws/conn.go

```diff
@@ -72,3 +85,15 @@ func (s *Server) getConns() []*conn {
 	}
 	return conns
 }
+
+func (s *Server) getClientConns(clientID string) []*conn {
+	s.mut.RLock()
+	defer s.mut.RUnlock()
```

Comment:
Maybe a similar issue here (and above)? But I'm not sure how hot this path is, so maybe it's not as important?